packages/api/lambdas/create-reconciliation-report.js

```diff
@@ -803,43 +838,56 @@ async function processRequest(params) {
   await reconciliationReportModel.create(reportRecord);
   log.info(`Report added to database as pending: ${JSON.stringify(reportRecord)}.`);
 
-  const env = params.env ? params.env : process.env;
-  const knex = await getKnexClient(env);
   const concurrency = env.CONCURRENCY || 3;
 
-  try {
-    const recReportParams = {
-      ...params,
-      createStartTime,
-      reportKey,
-      reportType,
-      knex,
-      concurrency,
-    };
-    log.info(`Beginning ${reportType} report with params: ${JSON.stringify(recReportParams)}`);
-    if (reportType === 'Internal') {
-      await createInternalReconciliationReport(recReportParams);
-    } else if (reportType === 'Granule Inventory') {
-      await createGranuleInventoryReport(recReportParams);
-    } else if (reportType === 'ORCA Backup') {
-      await createOrcaBackupReconciliationReport(recReportParams);
-    } else {
-      // reportType is in ['Inventory', 'Granule Not Found']
-      await createReconciliationReport(recReportParams);
-    }
-    await reconciliationReportModel.updateStatus({ name: reportRecord.name }, 'Generated');
-  } catch (error) {
-    log.error(`Error creating ${reportType} report ${reportRecordName}`, error);
-    log.error(errorify(error));
-    const updates = {
-      status: 'Failed',
-      error: {
-        Error: error.message,
-        Cause: errorify(error),
+  await pRetry(
+    async () => {
+      try {
+        const recReportParams = {
+          ...params,
+          createStartTime,
+          reportKey,
+          reportType,
+          knex,
+          concurrency,
+        };
+        log.info(`Beginning ${reportType} report with params: ${JSON.stringify(recReportParams)}`);
+        if (reportType === 'Internal') {
+          await createInternalReconciliationReport(recReportParams);
+        } else if (reportType === 'Granule Inventory') {
+          await createGranuleInventoryReport(recReportParams);
+        } else if (reportType === 'ORCA Backup') {
+          await createOrcaBackupReconciliationReport(recReportParams);
+        } else {
+          // reportType is in ['Inventory', 'Granule Not Found']
+          await createReconciliationReport(recReportParams);
+        }
+        await reconciliationReportModel.updateStatus({ name: reportRecord.name }, 'Generated');
+      } catch (error) {
+        if (error.message.includes('Connection terminated unexpectedly')) {
```

Comment:
If the error is a `Connection terminated unexpectedly` error, I will retry

Reply:
In which circumstances, there will be `Connection terminated unexpectedly` error?
Which db operation would throw the error?  Could we retry that operation instead of retry the report generation?
If the report generation is retried 2 times, will we get 3 reports or 1 report?